### ISSUE TITLE
Half of the samples crash

### ISSUE THREAD

**Issue by cheery** [SaschaWillems/Vulkan#33]
* bloom
 * deferred
 * displacement
 * geometryshader
 * instancing
 * mesh
 * occlusionquery
 * offscreen
 * parallaxmapping
 * pushconstants
 * radialblur
 * shadowmap
 * skeletalanimation
 * sphericalenvmapping
 * tesselation
 * texturecubemap
 * vulkanscene

GTX 960, Ubuntu 14.10

**Comment by cheery** [comment]
[vulkanscene.dump.txt](https://github.com/SaschaWillems/Vulkan/files/139402/vulkanscene.dump.txt)

Here's a vulkanscene dump. Maybe it tells you why it crashes and where?

**Comment by CapsAdmin** [comment]
Same thing happens to me using a gtx 970 on kubuntu 15.10 with neon packages (so newer Qt if that matters). I also had to install kernel 4.2.0-29-generic in order to use the driver.

If it helps here's deferred's vktrace [deferred.vktrace.zip](https://github.com/SaschaWillems/Vulkan/files/139517/deferred.vktrace.zip)
It doesn't crash when replaying it though.

**Comment by SaschaWillems** [comment]
Did you compile from source or are you using the prebuild binaries? Can you give me a call stack? From the list it seems that only samples using models (most of them) crash, so they may either not find the data folder or something is wrong with ASSIMP. If you compliled yourself, what ASSIMP version did you use?

**Comment by CapsAdmin** [comment]
I compiled everything from source with `cmake -DCMAKE_BUILD_TYPE=Debug ..`. 
This is an out of source build but I'm aware that the working directory needs to be Vulkan/bin/

If I attempt to run bloom outside of this working directory it will error like this
`bloom: /media/caps/ssd_840_120gb/Vulkan/base/vulkanTextureLoader.hpp:360: void vkTools::VulkanTextureLoader::loadCubemap(const char*, VkFormat, vkTools::VulkanTexture*): Assertion `!texCube.empty()' failed.`

Which is expected.

The "texture" sample manages to load the texture fine though. (Unrelated but the plane looks like it has a lod level is missing (guessing) cause it has a green shade in some angles)

**Comment by SaschaWillems** [comment]
Can you please run the demos with validation layers enabled and check for errors?

**Comment by CapsAdmin** [comment]
I set ENABLE_VALIDATION to true in bloom.cpp

```
caps@caps-MS-7798:/media/caps/ssd_840_120gb/Vulkan/bin$ ./bloom 
Searching the following paths for manifest files: /etc/vulkan/explicit_layer.d:/usr/share/vulkan/explicit_layer.d:

Searching the following paths for manifest files: /etc/vulkan/implicit_layer.d:/usr/share/vulkan/implicit_layer.d

Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_screenshot.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_object_tracker.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_api_dump.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_image.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_swapchain.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_vktrace_layer.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_device_limits.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_threading.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_mem_tracker.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_param_checker.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_draw_state.json, version "1.0.0"
Found manifest file /etc/vulkan/explicit_layer.d/VkLayer_unique_objects.json, version "1.0.0"
Searching the following paths for manifest files: /etc/vulkan/icd.d:/usr/share/vulkan/icd.d

Found manifest file /etc/vulkan/icd.d/nvidia_icd.json, version "1.0.0"
Searching for ICD drivers named libGL.so.1 default dir 

Build ICD instance extension list
Instance Extension: VK_KHR_surface (libGL.so.1) version 0.0.25
Instance Extension: VK_KHR_xcb_surface (libGL.so.1) version 0.0.6
Instance Extension: VK_KHR_xlib_surface (libGL.so.1) version 0.0.6
Instance Extension: VK_EXT_debug_report (libGL.so.1) version 0.0.1
Chain: instance: Loading layer library libVkLayer_unique_objects.so
Insert instance layer VK_LAYER_GOOGLE_unique_objects (libVkLayer_unique_objects.so)
Chain: instance: Loading layer library libVkLayer_image.so
Insert instance layer VK_LAYER_LUNARG_image (libVkLayer_image.so)
Chain: instance: Loading layer library libVkLayer_device_limits.so
Insert instance layer VK_LAYER_LUNARG_device_limits (libVkLayer_device_limits.so)
Chain: instance: Loading layer library libVkLayer_swapchain.so
Insert instance layer VK_LAYER_LUNARG_swapchain (libVkLayer_swapchain.so)
Chain: instance: Loading layer library libVkLayer_param_checker.so
Insert instance layer VK_LAYER_LUNARG_param_checker (libVkLayer_param_checker.so)
Chain: instance: Loading layer library libVkLayer_draw_state.so
Insert instance layer VK_LAYER_LUNARG_draw_state (libVkLayer_draw_state.so)
Chain: instance: Loading layer library libVkLayer_object_tracker.so
Insert instance layer VK_LAYER_LUNARG_object_tracker (libVkLayer_object_tracker.so)
Chain: instance: Loading layer library libVkLayer_mem_tracker.so
Insert instance layer VK_LAYER_LUNARG_mem_tracker (libVkLayer_mem_tracker.so)
Chain: instance: Loading layer library libVkLayer_threading.so
Insert instance layer VK_LAYER_LUNARG_threading (libVkLayer_threading.so)
Build ICD instance extension list
Instance Extension: VK_KHR_surface (libGL.so.1) version 0.0.25
Instance Extension: VK_KHR_xcb_surface (libGL.so.1) version 0.0.6
Instance Extension: VK_KHR_xlib_surface (libGL.so.1) version 0.0.6
Instance Extension: VK_EXT_debug_report (libGL.so.1) version 0.0.1
Device Extension: VK_KHR_swapchain (libGL.so.1) version 0.0.67
Device Extension: VK_NV_glsl_shader (libGL.so.1) version 0.0.1
device Chain: Increment layer reference count for layer library libVkLayer_unique_objects.so
Insert device layer VK_LAYER_GOOGLE_unique_objects (libVkLayer_unique_objects.so)
device Chain: Increment layer reference count for layer library libVkLayer_image.so
Insert device layer VK_LAYER_LUNARG_image (libVkLayer_image.so)
device Chain: Increment layer reference count for layer library libVkLayer_device_limits.so
Insert device layer VK_LAYER_LUNARG_device_limits (libVkLayer_device_limits.so)
device Chain: Increment layer reference count for layer library libVkLayer_swapchain.so
Insert device layer VK_LAYER_LUNARG_swapchain (libVkLayer_swapchain.so)
device Chain: Increment layer reference count for layer library libVkLayer_param_checker.so
Insert device layer VK_LAYER_LUNARG_param_checker (libVkLayer_param_checker.so)
device Chain: Increment layer reference count for layer library libVkLayer_draw_state.so
Insert device layer VK_LAYER_LUNARG_draw_state (libVkLayer_draw_state.so)
device Chain: Increment layer reference count for layer library libVkLayer_object_tracker.so
Insert device layer VK_LAYER_LUNARG_object_tracker (libVkLayer_object_tracker.so)
device Chain: Increment layer reference count for layer library libVkLayer_mem_tracker.so
Insert device layer VK_LAYER_LUNARG_mem_tracker (libVkLayer_mem_tracker.so)
device Chain: Increment layer reference count for layer library libVkLayer_threading.so
Insert device layer VK_LAYER_LUNARG_threading (libVkLayer_threading.so)
Segmentation fault (core dumped)
```

**Comment by CapsAdmin** [comment]
I'm not very efficent with c++ and debugging it but I've been looking into where it crashes in source (built with `cmake . -DCMAKE_BUILD_TYPE=Debug -G"CodeBlocks - Unix Makefiles"` ) and it crashes when you make a new VulkanMeshLoader. It doesn't have a constructor so I tried adding that to see if it would run it but it crashes before it runs the constructor.

Maybe it's something to do with our version of GCC or something? You asked if we ran the prebuilt binaries but I don't see those anywhere for linux.

**Comment by SaschaWillems** [comment]
What GCC version are you using? I have tested all examples on a Linux Mint 17.2 (NVIDIA GPU) and all of them ran fine.

64 Bit Linux binaries can be [found here](http://vulkan.gpuinfo.org/examples.php), though they're lagging behind the windows one, but I'll try to upload a new build this weekend.

**Comment by CapsAdmin** [comment]
I put the binaries in the bin dir and they work fine!

gcc -v says

```
caps@caps-MS-7798:/media/caps/ssd_840_120gb/Vulkan/bin$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/5/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 5.2.1-22ubuntu2' --with-bugurl=file:///usr/share/doc/gcc-5/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-5 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-5-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-5-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-5-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 5.2.1 20151010 (Ubuntu 5.2.1-22ubuntu2) 
```

**Comment by SaschaWillems** [comment]
Just tested this again on two different Linux machines. Lowest version tested was gcc 4.9.2 (Unbuntu 4.9.2-10ubuntu13)

Samples compile, link and run fine.

Do you have assimp installed globally? If so, what version?

**Comment by CapsAdmin** [comment]
I didn't initially. I tried to install it to see if that would fix it but it crashes before it uses assimp. 

I have /usr/lib/libassimp.so.3.0.1264

**Comment by SaschaWillems** [comment]
Should be fixed with PR #81
